### ISSUE TITLE
[game] Check for win and draw

### ISSUE THREAD

**Issue by heyapricot** [rvvergara/tictactoe#13]
A better way to check win a win or a draw IMHO is:

- Implement the methods in the `Board` class that check if all the elements in a row, column or diagonal are the same
- `Game` asks board:
```ruby
# game.rb
def my_method
  if board.has_sequence?
    # Game has ended
  elsif board.is_full?
    # Game is a draw
  end

  # Continue with the game
end
```

The checks belong to the `Board`  class. I strongly think this is the best implementation

**Comment by rvvergara** [comment]
Nice! Thanks.. We'll implement accordingly.

**Comment by rvvergara** [comment]
So [what we did](https://github.com/rvvergara/tictactoe/commit/43b2acf98bdddbd9748a318f15bfa8c0fa2ba005) was remove the actual logic of checking for combinations from `Game` and transferred it into `Board`. 

But we maintained the current way of checking for such winning combinations (in reference to #9, while we are still discussing it). At the very least while we are analyzing the best way to check for winning combinations ('hard coded implementation' vs dynamic checking you suggest in #9) we can quarantine this into the `Board` class. Game class now doesn't care how we know the player's moves are already a winning set.
